File: templates/problems/05_binary_search/04_find_min_rotated/go/find_min_rotated_test.go
package find_min_rotated

import "testing"

func TestFindMin(t *testing.T) {
	tests := []struct {
		name string
		nums []int
		want int
	}{
		{"rotated", []int{3, 4, 5, 1, 2}, 1},
		{"rotated once", []int{4, 5, 6, 7, 0, 1, 2}, 0},
		{"not rotated", []int{11, 13, 15, 17}, 11},
		{"single element", []int{1}, 1},
		{"two elements rotated", []int{2, 1}, 1},
		{"two elements sorted", []int{1, 2}, 1},
		{"min at end", []int{2, 3, 4, 5, 1}, 1},
		{"min at start", []int{1, 2, 3, 4, 5}, 1},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := findMin(tt.nums); got != tt.want {
				t.Errorf("findMin(%v) = %d, want %d", tt.nums, got, tt.want)
			}
		})
	}
}

File: templates/problems/05_binary_search/04_find_min_rotated/typescript/find_min_rotated.test.ts
import { describe, it, expect } from "vitest";
import { findMin } from "./find_min_rotated";

describe("findMin", () => {
  it("rotated", () => {
    expect(findMin([3, 4, 5, 1, 2])).toBe(1);
  });

  it("rotated once", () => {
    expect(findMin([4, 5, 6, 7, 0, 1, 2])).toBe(0);
  });

  it("not rotated", () => {
    expect(findMin([11, 13, 15, 17])).toBe(11);
  });

  it("single element", () => {
    expect(findMin([1])).toBe(1);
  });

  it("two elements rotated", () => {
    expect(findMin([2, 1])).toBe(1);
  });

  it("two elements sorted", () => {
    expect(findMin([1, 2])).toBe(1);
  });

  it("min at end", () => {
    expect(findMin([2, 3, 4, 5, 1])).toBe(1);
  });

  it("min at start", () => {
    expect(findMin([1, 2, 3, 4, 5])).toBe(1);
  });
});
